### PR TITLE
[8.0][mis_builder][ADD] mis_builder_analytic_filter

### PR DESCRIPTION
**Pull request by sbidoul**
Add analytic filtering on mis_builder reports. The filter is static (in the MIS Report instance configuration form) but also dynamic (in the preview and dashboard view) so the user can change analytic account on the fly.
- [x] Depends on #151 

image

### CODE REVIEW

**Comment by adrienpeiffer** [comment]
:+1: (Functionnal tests and code review). Nice improvement !

File: mis_builder_analytic_filter/views/mis_report_view.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+    <data>
+        <record model="ir.ui.view" id="mis_report_instance_view_form">
+            <field name="name">mis.report.instance.view.form (mis_builder_analytic_filter)</field>
+            <field name="model">mis.report.instance</field>
+            <field name="inherit_id" ref="mis_builder.mis_report_instance_view_form" />
+            <field name="arch" type="xml">
+                <field name='target_move' position="after">
+                    <field name="account_analytic_id" />
```
Comment (laetitia-gangloff) [inline mis_builder_analytic_filter/views/mis_report_view.xml]: I think you should add group : analytic.group_analytic_accounting

File: mis_builder_analytic_filter/static/src/xml/mis_widget.xml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<template>
+    <t t-extend="mis_builder.MisReport">
+        <t t-jquery=".oe_mis_builder_buttons" t-operation="prepend">
+            <button class="oe_mis_builder_generate_content"><img src="/web/static/src/img/icons/gtk-refresh.png"/> Refresh</button>
+        </t>
+        <t t-jquery=".oe_mis_builder_buttons" t-operation="after">
+            <div class="oe_mis_builder_analytic_axis" style="position: relative; display: inline-block;"></div>
```
Comment (laetitia-gangloff) [inline mis_builder_analytic_filter/static/src/xml/mis_widget.xml]: I think you should add group : analytic.group_analytic_accounting

Comment (sbidoul) [inline mis_builder_analytic_filter/static/src/xml/mis_widget.xml]: @laetitia-gangloff any idea how to implement this easily in the qweb widget?

Comment (laetitia-gangloff) [inline mis_builder_analytic_filter/static/src/xml/mis_widget.xml]: See here : https://github.com/OCA/account-financial-reporting/pull/170/commits/c39eb41d51f3333aa1e7a8a4ff4737e25484aa05

**Comment by sbidoul** [comment]
@laetitia-gangloff done and rebased

**Comment by laetitia-gangloff** [comment]
Thank you !
:+1:

**Comment by adrienpeiffer** [comment]
:+1: again. Thanks !

**Comment by sbidoul** [comment]
This one is ready for final review and merge.

File: mis_builder_analytic_filter/README.rst [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+============
+
+There is no specific installation procedure for this module.
+
+Configuration and Usage
+=======================
+
+In Accounting > Reporting > MIS Reports, there is a new field
+'Analytic Account' which can be used to filter move line on an
+analytic axis and it's children.
```
Comment (elicoidal) [inline mis_builder_analytic_filter/README.rst:22]: s/it's/their

File: mis_builder_analytic_filter/README.rst [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+can change the filter in the preview and dashboard views.
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/91/8.0
+
+Known issues / Roadmap
+======================
+
+While useful as is, we are aware that such analytic filtering may need
```
Comment (elicoidal) [inline mis_builder_analytic_filter/README.rst]: include in list with "*"

File: mis_builder_analytic_filter/README.rst [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+to be customized heavily depending on the customer context. This module can
+be extended or considered as an example.
+
+* at the moment, no filtering is done on other queries than move lines
```
Comment (elicoidal) [inline mis_builder_analytic_filter/README.rst]: s/at/At
Add period
or
Currently, only filtering on move line queries is performed.

File: mis_builder_analytic_filter/__openerp__.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    'name': "MIS Builder Analytic Axis Filter",
+    'version': '8.0.1.0.0',
+    'category': 'Reporting',
+    'summary': """
```
Comment (elicoidal) [inline mis_builder_analytic_filter/__openerp__.py:9]: This does not add any information compared with name. Why not use the README description?

File: mis_builder_analytic_filter/static/src/js/mis_builder.js [hunk 7]
```diff
@@ -0,0 +1,78 @@
+openerp.mis_builder_analytic_filter = function(instance) {
```
Comment (elicoidal) [inline mis_builder_analytic_filter/static/src/js/mis_builder.js:5]: shouldnot we need copyright in js files, following https://github.com/OCA/maintainer-tools/blob/master/template/module/static/src/js/module_name.js

**Comment by elicoidal** [comment]
some details. Other than that LGTM :+1:

File: mis_builder_analytic_filter/__openerp__.py (hunk 6 above)
Comment (sbidoul) [inline mis_builder_analytic_filter/__openerp__.py:9]: The description in the readme is not more informative either. It's still nicer to have a short summary that rephrases the module name. I'm open to suggestions.

File: mis_builder_analytic_filter/README.rst (hunk 3 above)
Comment (sbidoul) [inline mis_builder_analytic_filter/README.rst:22]: It's about the children of the analytic axis (singular) so I'd keep "its" (without the apostrophe). And I add a "s" to move line_s_.

**Comment by sbidoul** [comment]
@elicoidal thanks for the review.

**Comment by lmignon** [comment]
:+1:

File: mis_builder_analytic_filter/static/src/js/mis_builder.js (hunk 7 above)
Comment (sbidoul) [inline mis_builder_analytic_filter/static/src/js/mis_builder.js:5]: @elicoidal I added the copyright in the js files.

**Comment by sbidoul** [comment]
I handled @elicoidal's comments so I think this one is good for merging.

**Comment by sbidoul** [comment]
After further thinking, I set this in WIP, because for modularity, the refresh button must move to mis_buillder, and there should be a flag on mis.report.instance controlling if the analytic filter combo box appears in the widget or not

**Comment by andreschenkels** [comment]
Could this be merged. It works like a charm :+1:

**Comment by sbidoul** [comment]
@aschenkels-ictstudio yes we could merge this. Wel don't plan that the refactoring in v8 anymore.

I plan to work quite a bit on mis_builder in the coming month, so expect movement soon.

**Comment by hbrunn** [comment]
I PRed OCA/mis-builder#29, so I figure we can close this one

**Pull request closed without merging**